app/webpacker/components/RegistrationsV2/lib/RegistrationProvider.js

```diff
@@ -0,0 +1,60 @@
+import React, { createContext, useContext, useState } from 'react';
+import { useQuery } from '@tanstack/react-query';
+import { useDispatch } from '../../../lib/providers/StoreProvider';
+import { getSingleRegistration } from '../api/registration/get/get_registrations';
+import { setMessage } from '../Register/RegistrationMessage';
+
+const RegistrationContext = createContext();
+
+export default function RegistrationProvider({ competitionInfo, userInfo, children }) {
+  const dispatch = useDispatch();
+  const [isProcessing, setIsProcessing] = useState(false);
+
+  const {
+    data: registration,
+    isFetching,
+    refetch: refetchRegistration,
+  } = useQuery({
+    queryKey: ['registration', competitionInfo.id, userInfo.id],
+    queryFn: () => getSingleRegistration(userInfo.id, competitionInfo),
+    onError: (error) => {
+      dispatch(
+        setMessage(
+          `competitions.registration_v2.errors.${error?.response?.data?.json?.error || 'unknown'}`,
+          'negative',
+        ),
+      );
+    },
+  });
+
+  const isRegistered = Boolean(registration) && registration.competing.registration_status !== 'cancelled';
+  const isAccepted = isRegistered && registration.competing.registration_status === 'accepted';
+  const isRejected = isRegistered && registration.competing.registration_status === 'rejected';
+  const hasPaid = registration?.payment?.has_paid;
+
+  const value = {
+    isRegistered,
+    isAccepted,
+    isRejected,
+    hasPaid,
+    registration,
+    refetchRegistration,
+    isFetching,
+    isProcessing,
+    setIsProcessing,
```

Comment:
I'm not super familiar with the details here, but it seems weird to expose `setIsProcessing` to any component that wants it. It feels like this context/hook should be handling that internally and just expose `isProcessing`.

Possibly the query and effect in `Processing.jsx` could/should be moved here to accomplish that?

Reply:
mh.. that is true, my goal was to allow for it to be set by the backend as well (if someone refreshes), but that can be accomplished without exposing the isProcessing method